pyblish_nuke/lib.py

```diff
@@ -144,10 +144,88 @@ def add_to_filemenu():
 
     shortcut = os.environ.get("PYBLISH_HOTKEY", "")
 
-    cmd = "import pyblish_nuke;pyblish_nuke.show()"
+    cmd = "import pyblish_nuke;pyblish_nuke.publish()"
     menu.addCommand("Publish", cmd, shortcut, index=9)
+    cmd = "import pyblish_nuke;pyblish_nuke.show()"
+    menu.addCommand("Publish...", cmd, shortcut, index=10)
+
+    menu.addSeparator(index=11)
+
+
+class Splash(QtWidgets.QWidget):
+    """Splash screen for publishing."""
+
+    def __init__(self, parent=None):
+        super(Splash, self).__init__(parent)
+        self.setAttribute(QtCore.Qt.WA_DeleteOnClose)
+        self.setAttribute(QtCore.Qt.WA_TranslucentBackground)
+        self.setWindowFlags(
+            QtCore.Qt.WindowStaysOnTopHint |
+            QtCore.Qt.FramelessWindowHint
+        )
+
+        pixmap = QtGui.QPixmap(
+            os.path.join(os.path.dirname(__file__), "splash.png")
+        )
+        image = QtWidgets.QLabel()
+        image.setPixmap(pixmap)
+
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.addWidget(image)
+
+        self.bar = QtGui.QProgressBar()
+        layout.addWidget(self.bar)
+
+        # Center widget on screen
+        self.resize(100, 100)
+
+
+def publish():
+
+    splash = Splash()
+    splash.show()
+
+    def on_published(context):
+        try:
+            splash.close()
+        except RuntimeError:
+            # Splash already closed
+            pass
+
+        pyblish.api.deregister_callback(*callback)
```

Comment:
Better deregister this *before* closing the window, in case a different exception is thrown.